code/game/objects/items/stacks/tiles/tile_types.dm

```diff
@@ -20,9 +20,50 @@
 	drop_sound = 'sound/items/drop/axe.ogg'
 	pickup_sound = 'sound/items/pickup/axe.ogg'
 
+//crafting / welding vars
+	var/datum/material/material //*sigh* i guess this is how we're doing this.
+	var/craftable = FALSE //set to TRUE for tiles you can craft stuff from directly, like grass
+	var/can_weld = FALSE //set to TRUE for tiles you can reforge into their components via welding, like metal
+	var/welds_into = /obj/item/stack/material/steel //what you get from the welding. defaults to steel.
+	var/default_type = DEFAULT_WALL_MATERIAL
+
+
+
 /obj/item/stack/tile/Initialize()
 	. = ..()
 	randpixel_xy()
+	if(craftable)
+		material = get_material_by_name("[default_type]")
+		if(!material)
+			return INITIALIZE_HINT_QDEL
+		if(material) //sanity check
+			recipes = material.get_recipes()
+			stacktype = material.stack_type
+
+/obj/item/stack/tile/attackby(obj/item/W as obj, mob/user as mob)
+	..()
+	if (istype(W, /obj/item/weldingtool))
+		var/obj/item/weldingtool/WT = W
+
+		if(can_weld == FALSE)
+			to_chat("You can't reform these into their original components.")

```

Comment:
Missing return

Reply:
github refuses to acknowledge this exists now but it totally does